devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy

```diff
@@ -133,7 +133,7 @@ def parse(String description) {
  	
     //check in configuration change
     if (!device.latestValue('configuration')) result = poll()
-    if (device.latestValue('configuration').toInteger() != interval && interval != null) {  	
+    if (device.latestValue('configuration') as float != interval && interval != null) {
```

Comment:
Comparisons of equality with floats don't seem like a great idea. Is this ever not an integer?

Reply:
log messages show it as being `"10.0"`, which shockingly does not cast correctly.

Reply:
Line 320 has `device.latestValue('configuration').toInteger()` as well. Is the exception around `device.latestValue('configuration')` being Null and trying to call toInteger() on it? In which case, `device.latestValue('configuration')?.toInteger()` is probably the safest route.

Also, I suspect that "configuration" needs to be defined as an attribute in the metadata block.

`attribute "configuration", "number"`

Reply:
the log message is a NumberFormatException, not an NPE. thanks for pointing out that other occurence

Reply:
"configuration" is a preference